Derive global map of type deserializers (#2814)
anoma/namada: .changelog/unreleased/improvements/2814-global-deserializer-map.md

 - This PR adds a proc macro that registers the deserializer of a type in a hashmap. This allows us to verify that 

>    data blob deserializes as correctly if where are in possession of the hash map key. ([\#2814](https://github.com/anoma/namada/pull/2814))

Reviewer: ```suggestion
   data blob deserializes correctly if we are in possession of the hash map key. ([\#2814](https://github.com/anoma/namada/pull/2814))
```
